apps/web/components/thread/thread-item.tsx

```
@@ -0,0 +1,422 @@
+  }, [threadItem.goals, threadItem.steps]);
+
+  const hasAnswer = useMemo(() => {
+    return threadItem.answer?.text && threadItem.answer?.text.length > 0;
+  }, [threadItem.answer]);
+
+  const hasClarifyingQuestions = useMemo(() => {
+    return threadItem.answer?.object?.clarifyingQuestion && threadItem.answer?.objectType === 'clarifyingQuestions';
+  }, [threadItem.answer]);
+
+  const allSources = useMemo(() => {
+    const sourceMatches = threadItem.answer?.text?.match(/<Source>([^]*?)<\/Source>/g) || [];
```

Biome (1.9.4)

[error] The regular expression includes this negated empty character class.

Negated empty character classes match anything.
If you want to match against [, escape it \[.
Otherwise, remove the character class or fill it.

(lint/correctness/noEmptyCharacterClassInRegex)